File: wowsca/src/main/java/jwt/validation/wowsca/model/ViewDG.java
package jwt.validation.wowsca.model;

import java.util.ArrayList;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ViewDG {
    private int codigo;
    private String nome;
    private String descricao;
    private String data;
    private double valor;
    private ArrayList<Parte> partes;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    private class Parte{
        private double valor;
        private String nome;
        private boolean pago;
    }
}

File: Front/src/app/model/viewDG.ts
import { Parte } from "./parte";

export class ViewDG{
  codigo?: number;
  nome?: string;
  descricao?: string;
  data?: string;
  valor?: number;
  partes?: Parte[];
  constructor(codigo?: number, nome?: string, descricao?: string, data?: string, valor?: number, partes?: Parte[]){
    this.codigo = codigo;
    this.nome = nome;
    this.descricao = descricao;
    this.data = data;
    this.valor = valor;
    this.partes = partes;
  }
}
